### PR TITLE
OCPBUGS-2513: Set PublicContainerRegistries for entries in mirror

### PR DESCRIPTION
**Pull request by bfournie**
The PublicContainerRegistries env variable is used by assisted-service so pull-secret entries aren't needed for external locations in a disconnected environment. Currently this is set to "quay.io". This change sets it to the locations defined in registries.conf to avoid adding pull-secret entries.

See https://bugzilla.redhat.com/show_bug.cgi?id=1990659 for reference.

### CODE REVIEW

**Review by zaneb — COMMENTED** [review]
Makes sense. So PUBLIC_CONTAINER_REGISTRIES is the list of stuff that is exempt from the validation that they are referenced in the pull secret. Because we know that anything that is mirrored may just get pulled from the mirror instead, we add those registries to the list to avoid complaints.
/approve

File: pkg/asset/agent/image/ignition.go [hunk 1]
```diff
@@ -459,3 +464,27 @@ func getMirrorFromRelease(releaseImage string, registriesConfig *mirror.Registri
...
+				location = config.Location[:i]
+			} else {
+				location = config.Location
+			}
+			if !strings.Contains(sb.String(), location) {
+				if sb.String() != "" {
+					sb.WriteByte(',')
+				}
+				sb.WriteString(location)
+			}
```
[the remark this answers is not in the document]
Comment (bfournie) [reply pkg/asset/agent/image/ignition.go:483]: > nit: I would probably have added the strings to a slice and then used `strings.Join`.

Yeah that did save a couple lines. Updated.

**Pull request merged**